tracer/src/Datadog.Trace/IAST/Propagation/StringModuleImpl.cs

```diff
@@ -386,6 +386,103 @@ private static int GetPositionAndUpdateRangesInStringJoin(TaintedObjects tainted
         return pos;
     }
 
+    /// <summary> OnStringTrim with single trimchar </summary>
+    /// <param name="self"> Param 1 </param>
+    /// <param name="result"> Result </param>
+    /// <param name="trimChar"> the trim char, null for char.IsWhiteSpace(self[indexLeft]) </param>
+    /// <param name="left"> Apply left trim </param>
+    /// <param name="right"> Apply right trim </param>
+    public static string? OnStringTrim(string self, string result, char? trimChar, bool left, bool right)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(result))
```

Comment:
Should you also handle the case where result equals self? (in case the customers trim a char that isn't present in the string)?

Reply:
We can check ReferenceEquals(self, result). When trim() does not change the input string, the result has the same reference, so this could be added to exclude the checking of that case, which would increase the performance. I would leave ReferenceEquals(self, result) better than a regular == comparison just in case a particular .Net version in the future changes that behavior. I have updated the code.